### PR TITLE
[MM-42565] Toast - Follow/Unfollow thread with undo functionality

### PR DESCRIPTION
**Pull request by tanmaythole**
#### Summary

This PR adds the snack bar whenever the user follows/unfollows a thread. On top of that, the user is able to undo the previous action.

#### Ticket Link

https://github.com/mattermost/mattermost-server/issues/21241

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [X] Has UI changes
- [X] Includes text changes and localization file updates
- [X] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: Android 12 - One UI Core 4.1

#### Screenshots

https://user-images.githubusercontent.com/72058456/229788750-293e448c-7096-4637-ab03-60af9882c530.mp4

#### Release Note

```release-note
Users can follow or unfollow a thread and undo the action if needed.
```

### CODE REVIEW

**Review by enahum — CHANGES_REQUESTED** [review]
two minor comments, other than that LGTM

File: app/actions/remote/thread.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 import {markTeamThreadsAsRead, markThreadAsViewed, processReceivedThreads, switchToThread, updateTeamThreadsSync, updateThread} from '@actions/local/thread';
 import {fetchPostThread} from '@actions/remote/post';
+import {showThreadFollowingSnackbar} from '@app/utils/snack_bar';
```
[the remark this answers is not in the document]
Comment (tanmaythole) [reply app/actions/remote/thread.ts]: Done 👍🏻

File: app/actions/remote/thread.ts [hunk 2]
```diff
@@ -202,7 +203,7 @@ export const markThreadAsUnread = async (serverUrl: string, teamId: string, thre
     }
 };
 
-export const updateThreadFollowing = async (serverUrl: string, teamId: string, threadId: string, state: boolean) => {
+export const updateThreadFollowing = async (serverUrl: string, teamId: string, threadId: string, state: boolean, showSnackBar = false) => {
```
[the remark this answers is not in the document]
Comment (tanmaythole) [reply app/actions/remote/thread.ts]: Done 👍🏻

**Review by enahum — APPROVED** [review]

**Comment by enahum** [comment]
@tanmaythole this file needs changing app/screens/thread/thread_follow_button/thread_follow_button.tsx:59:9

**Review by enahum — DISMISSED** [review]

**Comment by matthewbirtch** [comment]
> Do we not require a snack bar in the thread screen after following/unfollowing?
>> Do we? that would be odd to me, but I'll defer to @matthewbirtch to answer that question, for now just set it to false as that is how it used to be even after you changes.

I don't think we do since the follow/unfollow button has a visible change

**Comment by enahum** [comment]
> Looks good @tanmaythole. The only thing I notice is that the `Undo` button is too close to the right edge of the toast. Can you add 6 padding on the right side of the `Undo` button so there is a bit more breathing room on the right side of the toast?
> 
> 
> 
> image
> 
> 

That would affect all snackbars with undo, is that expected?

**Comment by matthewbirtch** [comment]
> > Looks good @tanmaythole. The only thing I notice is that the `Undo` button is too close to the right edge of the toast. Can you add 6 padding on the right side of the `Undo` button so there is a bit more breathing room on the right side of the toast?
> > image
> 
> That would affect all snackbars with undo, is that expected?

Yes @enahum. I'm not sure if the position of the undo here has changed or if it just wasn't noticed before, but it's too close to the edge

**Review by matthewbirtch — APPROVED** [review]
Looks good to me!

**Review by enahum — APPROVED** [review]

**Comment by enahum** [comment]
@M-ZubairAhmed can you review this PR, I would like to include this in today's beta build.

**Review by M-ZubairAhmed — APPROVED** [review]

**Pull request merged**